Source/WebKit/UIProcess/ApplicationStateTracker.mm

```diff
@@ -167,10 +167,33 @@ ApplicationType applicationType(UIWindow *window)
     setScene(nil);
     setViewController(nil);
 
+    removeAllObservers();
+
     allApplicationStateTrackers().remove(*this);
     updateApplicationBackgroundState();
 }
 
+void ApplicationStateTracker::removeAllObservers()
+{
+    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
```

Comment:
We should use a RetainPtr.

Reply:
`defaultCenter` is a singleton object; it's guaranteed to be the same object every time `+defaultCenter` is called, so no retain is needed.

Reply:
Should I point you to the smart pointer guidelines?

Reply:
The times of "trust me it's OK in this case" are long gone. We have a set of smart pointers that we apply consistently, no matter what.